src/vs/platform/terminal/node/terminalProcess.ts

```diff
@@ -111,15 +120,24 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 			conptyInheritCursor: useConpty && !!_shellLaunchConfig.initialText
 		};
 		// Delay resizes to avoid conpty not respecting very early resize calls
```

Comment:
Move this down one line so it's clear it's about `if (useConpty && cols ===...`

Another small comment explaining the `WindowsShellHelper` conditional below wouldn't hurt either.

Reply:
moved it down a line. not sure which `WindowsShellHelper` conditional you mean

Reply:
This is what I meant:

```ts
		if (platform.isWindows) {
			// Delay resizes to avoid conpty not respecting very early resize calls
			if (useConpty && cols === 0 && rows === 0 && this._shellLaunchConfig.executable?.endsWith('Git\\bin\\bash.exe')) {
				this._delayedResizer = new DelayedResizer();
				this._register(this._delayedResizer.onTrigger(dimensions => {
					this._delayedResizer?.dispose();
					this._delayedResizer = undefined;
					if (dimensions.cols && dimensions.rows) {
						this.resize(dimensions.cols, dimensions.rows);
					}
				}));
			}
			// WindowsShellHelper is used to fetch the process title and shell type
			this.onProcessReady(e => {
				this._windowsShellHelper = this._register(new WindowsShellHelper(e.pid));
				this._register(this._windowsShellHelper.onShellTypeChanged(e => this._onProcessShellTypeChanged.fire(e)));
				this._register(this._windowsShellHelper.onShellNameChanged(e => this._onProcessTitleChanged.fire(e)));
			});
		}
```